### PR TITLE
On merge of already closed PR redirect back to the pulls page

### PR DESCRIPTION
**Pull request by zeripath**

### CODE REVIEW

**Review by techknowlogick — APPROVED** [review]

**Review by jolheiser — APPROVED** [review]

**Comment by zeripath** [comment]
It seemed simplest to just redirect back with the message "The pull request has been closed."

We could be cleverer with double checking that it is a pull etc. but I dunno if it is worth it.

File: routers/repo/pull.go [hunk 1]
```diff
@@ -679,7 +679,12 @@ func MergePullRequest(ctx *context.Context, form auth.MergePullRequestForm) {
 		return
 	}
 	if issue.IsClosed {
-		ctx.NotFound("MergePullRequest", nil)
+		if issue.IsPull {
+			ctx.Flash.Error(ctx.Tr("repo.pulls.is_closed"))
+		} else {
+			ctx.Flash.Error(ctx.Tr("repo.issues.closed_title"))
+		}
+		ctx.Redirect(ctx.Repo.RepoLink + "/pulls/" + com.ToStr(issue.Index))
```
Comment (6543) [inline routers/repo/pull.go]: Why not set issue/pull specific redirect **in** if statement?

Comment (6543) [reply routers/repo/pull.go]: I know it works without - but if you read throu the code this is weared

File: routers/repo/pull.go [hunk 2]
```diff
@@ -679,7 +679,13 @@ func MergePullRequest(ctx *context.Context, form auth.MergePullRequestForm) {
 		return
 	}
 	if issue.IsClosed {
-		ctx.NotFound("MergePullRequest", nil)
+		if issue.IsPull {
+			ctx.Flash.Error(ctx.Tr("repo.pulls.is_closed"))
+			ctx.Redirect(ctx.Repo.RepoLink + "/issues/" + com.ToStr(issue.Index))
```
Comment (6543) [inline routers/repo/pull.go]:
```suggestion
			ctx.Redirect(ctx.Repo.RepoLink + "/pulls/" + com.ToStr(issue.Index))
```

File: routers/repo/pull.go [hunk 3]
```diff
@@ -679,7 +679,13 @@ func MergePullRequest(ctx *context.Context, form auth.MergePullRequestForm) {
...
+			return
+		}
+		ctx.Flash.Error(ctx.Tr("repo.issues.closed_title"))
+		ctx.Redirect(ctx.Repo.RepoLink + "/pulls/" + com.ToStr(issue.Index))
```
Comment (6543) [inline routers/repo/pull.go]:
```suggestion
		ctx.Redirect(ctx.Repo.RepoLink + "/issues/" + com.ToStr(issue.Index))
```

**Review by 6543 — CHANGES_REQUESTED** [review]
thanks ... but you switched something :D

File: routers/repo/pull.go (hunk 1 above)
Comment (zeripath) [reply routers/repo/pull.go]: Done

**Comment by zeripath** [comment]
It appears that I am too full of cold to code!

**Review by 6543 — APPROVED** [review]

**Pull request merged**